src/modules/royalty.ts

```diff
@@ -0,0 +1,152 @@
+import { Royalty, Royalty__factory } from "@3rdweb/contracts";
+import { BigNumberish } from "ethers";
+import { ModuleType, Role, RolesMap } from "../common";
+import { Currency, getCurrencyMetadata } from "../common/currency";
+import { Module } from "../core/module";
+
+export interface IRoyaltyModule {
+  /**
+   * Get the total shares held by payees
+   */
+  totalShares(): Promise<BigNumberish>;
+
+  /**
+   * Get the total amount of royalty already released
+   */
+  totalReleased(): Promise<BigNumberish>;
+
+  /**
+   * Returns the amount of `token` released to payees
+   *
+   * @param tokenAddress - The ERC20 token address to check royalty distributions for
+   */
+  totalReleasedByToken(tokenAddress: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of shares held by an account at the address `address`
+   *
+   * @param address - The address to check the shares held for
+   */
+  shares(address: string): Promise<BigNumberish>;
```

Comment:
what are shares? Am I investing in a royalty? Can I buy them on a secondary market?